lib/resultset.js

```diff
@@ -43,6 +49,55 @@ function getRows() {
 
 var getRowsPromisified = util.promisify(getRows);
 
+// seed batch size with an arbitrary number to prevent unwitting hogging of the main event loop
+function nonBlockingGetRows(){
+  let numRows=null;
+  let batchSize=null;
+  let callback=null;
+  const args = [].slice.call(arguments);
+
+  // parse the args, and maintain callback support
+  for(let i = 0; i < args.length; i++) {
+    if(typeof args[i] === 'number' && numRows === null) {
+      numRows = args[i];
+    } else if(typeof args[i] === 'number' && batchSize === null) {
+      batchSize = args[i];
+    } else if (typeof args[i] === 'function') {
+      callback = args[i];
+    } else {
+      console.error(`unrecognized argument ${args[i]} of type ${typeof args[i]}`)
+    }
+  }
+
+  // default values
+  numRows = numRows ?? 0;
+  batchSize = batchSize ?? 1000;
```

Comment:
So does this imply that if a batchSize is not supplied, then it will use 1000, is there a way to say no batch size, so we get no pauses?  It is not critical because you provided the env to choose the original getRows vs the new non blocking version.